normandy/recipes/storage.py

```diff
@@ -0,0 +1,29 @@
+import json
+
+from django.db import transaction
+
+from product_details.storage import PDFileStorage
+
+
+class ProductDetailsRelationalStorage(PDFileStorage):
+    """
+    Extends the in-database storage for product_details to provide a
+    database table of locales for other models to have foreign keys to.
+    """
+    @transaction.atomic
+    def update(self, name, content, last_modified):
```

Comment:
Did this have any tests before? If not, it seems pretty testable in terms of the API.